From 19ff17fce901afcc6c9be29509e28e53a5d71530 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Wed, 14 Apr 2010 09:06:04 -0700
Subject: [PATCH 23/25] XQuartz: Fix possible NULL dereference in ListenOnOpenFD

<rdar://problem/7862319>

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
Reviewed-by: Marc Majka <majka@apple.com>
(cherry picked from commit fe7778e58e099d353689755ed2f5aa440569ebe3)
---
 os/connection.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git xorg-server-1.8.0/os/connection.c xorg-server-1.8.0/os/connection.c
index 3ff93bb..61ba72a 100644
--- xorg-server-1.8.0/os/connection.c
+++ xorg-server-1.8.0/os/connection.c
@@ -1253,10 +1253,11 @@ MakeClientGrabPervious(ClientPtr client)
 void ListenOnOpenFD(int fd, int noxauth) {
     char port[256];
     XtransConnInfo ciptr;
+    const char *display_env = getenv("DISPLAY");
 
-    if(!strncmp(getenv("DISPLAY"), "/tmp/launch", 11)) {
+    if(display_env && (strncmp(display_env, "/tmp/launch", 11) == 0)) {
         /* Make the path the launchd socket if our DISPLAY is set right */
-        strcpy(port, getenv("DISPLAY"));
+        strcpy(port, display_env);
     } else {
         /* Just some default so things don't break and die. */
         sprintf(port, ":%d", atoi(display));
-- 
1.7.0.4

